packages/format-library/src/link/inline.js

```diff
@@ -227,7 +227,7 @@ class InlineLinkUI extends Component {
 						onKeyPress={ stopKeyPropagation }
 						url={ url }
 						onEditLinkClick={ this.editLink }
-						linkClassName={ isValidHref( prependHTTP( url ) ) ? undefined : 'has-invalid-link' }
```

Comment:
I think we should fix `prependHTTP` instead. 

```js
* @param {string} url The URL to test.
```

In its definition, it's expected to pass `string` so it should return the original value otherwise. It'd be similar to what `addQueryArgs` does.

Reply:
Would you prefer the fix to prependHTTP as a separate PR to the UI enhancements that @senadir has suggested?

Reply:
UI enhancement can be filed as a separate issue. We should fix the bug in the first place to make it usable :)

Reply:
I tested it with the Class block and it looks like the way it's handled there is, it gets never converted into a link if you leave the input field empty.

Reply:
I have updated the PR to just return the original value in prependHTTP instead. This fixes the bug - not sure if we want to do a proper check for a string value rather than a simple truthy check. 

With removing the link if url is not set, this is slightly complicated by the fact that we need to account for existing named anchor tags that will have no url attribute. The Classic block does account for these, and it has a different UI treatment for anchors. Do we want to combine looking at this in a separate PR along with the UI enhancements mentioned above?

Reply:
My feeling is that it's probably best to get this fix in as is, and then look to do other improvements separately. The current fix improves things quite a bit, and leaves only a couple of rough edges (which is way better than a crashing block).

Bearing in mind there's some work (https://github.com/WordPress/gutenberg/pull/17846) to revamp the link interface going on separately, it might be best to create a separate issue and seek some design feedback on how to handle this use case. It can then be factored in to the new design.

What do think @gziolo?

Reply:
Filed a follow-up: https://github.com/WordPress/gutenberg/issues/17972